### PR TITLE
Add information in the findContours documentation.

### PR DESCRIPTION
**Pull request by catree**
I propose to add in the findContours documentation a way to retrieve the contours located at the border.
I propose to add also the type (`std::vector<std::vector<cv::Point> >` for contours and `std::vector<cv::Vec4i>` for hierarchy) that should suit to most of the users.

Related issues #4374 and #4988. 

Edit:
the following code should highlight the "issue" at the border and the proposed workaroung:
 ```
cv::Mat img = cv::Mat::ones(8, 10, CV_8U);
std::cout << "img:\n" << img << std::endl;

cv::Mat img_with_border;
cv::copyMakeBorder(img, img_with_border, 1, 1, 1, 1, cv::BORDER_CONSTANT, cv::Scalar(0));
std::cout << "\nimg_with_border:\n" << img_with_border << std::endl;

std::vector<std::vector<cv::Point> > contours;
cv::findContours(img, contours, cv::RETR_LIST, cv::CHAIN_APPROX_NONE);

cv::Mat img_draw_contours = cv::Mat::zeros(img.size(), CV_8U);
for (size_t cpt = 0; cpt < contours.size(); cpt++) {
  cv::drawContours(img_draw_contours, contours, cpt, cv::Scalar(255));
}
std::cout << "\nimg_draw_contours:\n" << img_draw_contours << std::endl;

std::vector<std::vector<cv::Point> > contours_border;
cv::findContours(img_with_border, contours_border, cv::RETR_LIST, cv::CHAIN_APPROX_NONE, cv::Point(-1,-1));

cv::Mat img_draw_contours_border = cv::Mat::zeros(img.size(), CV_8U);
for (size_t cpt = 0; cpt < contours_border.size(); cpt++) {
  cv::drawContours(img_draw_contours_border, contours_border, cpt, cv::Scalar(255));
}
std::cout << "\nimg_draw_contours_border:\n" << img_draw_contours_border << std::endl;
```

Corresponding output:
```
img:
[  1,   1,   1,   1,   1,   1,   1,   1,   1,   1;
   1,   1,   1,   1,   1,   1,   1,   1,   1,   1;
   1,   1,   1,   1,   1,   1,   1,   1,   1,   1;
   1,   1,   1,   1,   1,   1,   1,   1,   1,   1;
   1,   1,   1,   1,   1,   1,   1,   1,   1,   1;
   1,   1,   1,   1,   1,   1,   1,   1,   1,   1;
   1,   1,   1,   1,   1,   1,   1,   1,   1,   1;
   1,   1,   1,   1,   1,   1,   1,   1,   1,   1]

img_with_border:
[  0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   1,   1,   1,   1,   1,   1,   1,   1,   1,   1,   0;
   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0,   0]

img_draw_contours:
[  0,   0,   0,   0,   0,   0,   0,   0,   0,   0;
   0, 255, 255, 255, 255, 255, 255, 255, 255,   0;
   0, 255,   0,   0,   0,   0,   0,   0, 255,   0;
   0, 255,   0,   0,   0,   0,   0,   0, 255,   0;
   0, 255,   0,   0,   0,   0,   0,   0, 255,   0;
   0, 255,   0,   0,   0,   0,   0,   0, 255,   0;
   0, 255, 255, 255, 255, 255, 255, 255, 255,   0;
   0,   0,   0,   0,   0,   0,   0,   0,   0,   0]

img_draw_contours_border:
[255, 255, 255, 255, 255, 255, 255, 255, 255, 255;
 255,   0,   0,   0,   0,   0,   0,   0,   0, 255;
 255,   0,   0,   0,   0,   0,   0,   0,   0, 255;
 255,   0,   0,   0,   0,   0,   0,   0,   0, 255;
 255,   0,   0,   0,   0,   0,   0,   0,   0, 255;
 255,   0,   0,   0,   0,   0,   0,   0,   0, 255;
 255,   0,   0,   0,   0,   0,   0,   0,   0, 255;
 255, 255, 255, 255, 255, 255, 255, 255, 255, 255]
```

### CODE REVIEW

File: modules/core/include/opencv2/core.hpp [hunk 1]
```diff
@@ -300,19 +300,19 @@ function does not copy src itself but simply constructs the border, for example:
 @endcode
 @note When the source image is a part (ROI) of a bigger image, the function will try to use the
 pixels outside of the ROI to form a border. To disable this feature and always do extrapolation, as
-if src was not a ROI, use borderType | BORDER_ISOLATED.
+if src was not a ROI, use borderType | cv::BORDER_ISOLATED.
```
Comment (vpisarev) [inline modules/core/include/opencv2/core.hpp]: I'm against the overuse of `cv::`. In many OpenCV samples we write `using namespace cv;` to make the code more clean. And let's keep it clean

Comment (catree) [inline modules/core/include/opencv2/core.hpp]: In fact, in this case I added `cv::` to be able to click on the word in the documentation generated by doxygen. But I'm fine to remove it if you want.

Comment (alalek) [inline modules/core/include/opencv2/core.hpp]: `cv::` for automatic links generation is widely used now ([example](https://github.com/opencv/opencv/blob/4ed40fd6946269ea36c58e5c76a277e7a871c269/modules/core/include/opencv2/core.hpp#L257))

BTW, `#BORDER_ISOLATED` should also create proper link, but IMHO "cv::" prefix looks more natural than "#".

**Comment by sovrasov** [comment]
@catree we decided to integrate the workaround into findContours. Look at #7516.

**Comment by catree** [comment]
Close this PR as it is replaced by #7516.

**Pull request closed without merging**